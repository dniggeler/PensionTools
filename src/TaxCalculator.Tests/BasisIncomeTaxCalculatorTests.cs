using System.Threading.Tasks;
using FluentAssertions;
using PensionCoach.Tools.TaxCalculator.Abstractions;
using PensionCoach.Tools.TaxCalculator.Abstractions.Models;
using Snapshooter.Xunit;
using Xunit;

namespace TaxCalculator.Tests
{
    [Trait("Calculator", "Income Tax")]
    public class BasisIncomeTaxCalculatorTests : IClassFixture<TaxCalculatorFixture<IBasisIncomeTaxCalculator>>
    {
        private readonly TaxCalculatorFixture<IBasisIncomeTaxCalculator> _fixture;

        public BasisIncomeTaxCalculatorTests(TaxCalculatorFixture<IBasisIncomeTaxCalculator> fixture)
        {
            _fixture = fixture;
        }

        [Fact(DisplayName = "Basis Income Tax")]
        public async Task ShouldCalculateBasisIncomeTax()
        {
            // given
            int calculationYear = 2018;

            var taxPerson = new BasisTaxPerson
            {
                Canton = "ZH",
                CivilStatus = CivilStatus.Married,
                TaxableAmount = 99995
            };

            // when
            var result = await _fixture.Calculator.CalculateAsync(calculationYear, taxPerson);

            result.IsRight.Should().BeFalse();
            Snapshot.Match(result);
        }
    }
}
