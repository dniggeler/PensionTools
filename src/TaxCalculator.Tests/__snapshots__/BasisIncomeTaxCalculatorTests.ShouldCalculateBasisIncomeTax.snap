[
  {
    "State": "IsLeft",
    "Right": null,
    "Left": {
      "CalculationYear": 2018,
      "DeterminingFactorTaxableAmount": 99900.0,
      "TaxAmount": 4878.0
    }
  }
]
